src/adagrams.js

```diff
@@ -1,15 +1,140 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+
+function getRandomProperty() {
+  const keys = Object.keys(LETTER_POOL);
+
+  return keys[Math.floor(Math.random() * keys.length)];
+}
+
+let drawnLetters = [];
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  while (drawnLetters.length < 10) {
+    let myCurrentKey = getRandomProperty();
+    let myCurrentValue = LETTER_POOL[myCurrentKey];
+    if (myCurrentValue !== 0) {
+      drawnLetters.push(myCurrentKey);
+      LETTER_POOL[myCurrentKey] = myCurrentValue - 1;
+    }
+    if (LETTER_POOL[myCurrentKey] === 0) {
+      delete LETTER_POOL[myCurrentKey];
+    }
+  }
+  
+  return drawnLetters;
 };
 
+
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+  let bankCopy = JSON.parse(JSON.stringify(lettersInHand))
+  for (let letter of input) {
+    if (!bankCopy.includes(letter.toUpperCase())){
+      return false
+    } else if (bankCopy.includes(letter.toUpperCase())) {
+      bankCopy = bankCopy.filter(value => value !== letter)
+      }
+    }
+    return true
+  };
+  
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+  let score = 0;
+  let lengthOfWord = word.length;
+  const letterValueDict = { 
+     A: 1, E: 1, I: 1, O: 1, U: 1, L: 1,
+     N: 1, R: 1, S: 1, T: 1, D: 2, G: 2,
+     B: 3, C: 3, M: 3, P: 3, F: 4, V: 4,
+     W: 4, Y: 4, K: 5, J: 8, X: 8, Q: 10,
+     Z: 10, H:4
+    };
+
+    if (word.length === 0){
+      return 0;
+    }
+
+    if(lengthOfWord >= 7){
```

Comment:
Best practice is to place a space after the if keyword to avoid looking like a function call!